File: python/pytorch/train/resnet18/train_fashion.py
from python.pytorch.models.resnet18 import run_experiment

if __name__ == "__main__":
    run_experiment(
        dataset_path="data/fashion_mnist_png",
        output_file="resnet18_fashion.csv",
        checkpoint_path="checkpoints/resnet18_fashion.pth",
        img_size=(32, 32),
        grayscale=True
    )

# PYTHONPATH=. python3 python/pytorch/train/resnet18/train_fashion.py

File: R/train/resnet18/train_fashion.r
library(torch)
library(torchvision)

source("R/models/resnet18.r")

cat("Script avviato!\n")

run_experiment(
    dataset_path = "data/fashion_mnist_png", 
    checkpoint_path = "R/checkpoints/resnet18_fashion_r.pt",
    img_size = c(32, 32), 
    grayscale = TRUE, # Fashion-MNIST è monocromatico → lo forziamo a 3 canali
    test_split = "test"
)